nemo/collections/nlp/models/question_answering/qa_model.py

```diff
@@ -184,29 +184,42 @@ def inference(
         self,
         file: str,
         batch_size: int = 1,
+        num_samples: int = -1,
         output_nbest_file: Optional[str] = None,
         output_prediction_file: Optional[str] = None,
-    ) -> List[int]:
+    ):
         """
-        Get prediction for the queries
+        Get prediction for unlabeled inference data
         Args:
-            queries: text sequences
+            file: inference data
             batch_size: batch size to use during inference
-            max_seq_length: sequences longer than max_seq_length will get truncated. default -1 disables truncation.
+            num_samples: number of samples to use of inference data. Default: -1 if all data should be used.
+            output_nbest_file: optional output file for writing out nbest list
+            output_prediction_file: optional output file for writing out predictions
         Returns:
-            all_preds: model predictions
+            all_predictions: model predictions
+            all_nbest: model nbest list
         """
         # store predictions for all queries in a single list
         all_predictions = []
         all_nbest = []
         mode = self.training
-        device = next(self.parameters()).device
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
         try:
             # Switch model to evaluation mode
             self.eval()
+            self.to(device)
```

Comment:
Do we need to move the model to another device?

Reply:
this is required in case since there is a weird behavior of PTL. I followed token_classification_model.
The model is shifted to cpu after training on gpu. So if you do inference afterwards this is required